src/generic/stage2/tape_builder.h

```diff
@@ -177,12 +177,11 @@ simdjson_warn_unused simdjson_really_inline error_code tape_builder::visit_root_
   // practice unless you are in the strange scenario where you have many JSON
   // documents made of single atoms.
   //
-  uint8_t *copy = static_cast<uint8_t *>(malloc(iter.remaining_len() + SIMDJSON_PADDING));
-  if (copy == nullptr) { return MEMALLOC; }
-  std::memcpy(copy, value, iter.remaining_len());
-  std::memset(copy + iter.remaining_len(), ' ', SIMDJSON_PADDING);
-  error_code error = visit_number(iter, copy);
-  free(copy);
+  std::unique_ptr<uint8_t[]>copy(new (std::nothrow) uint8_t[iter.remaining_len() + SIMDJSON_PADDING]);
```

Comment:
Note that On Demand has a version of this that uses the stack instead, by restricting the number of digits you can have in a number / bool and tossing an error if you exceed it. get_root_double() and friends. It's not ideal since floating-point numbers can in fact exceed this limit, but I capped it generously at the limit of possible significant digits, and root floating-point numbers are pretty uncommon to boot.

I wouldn't spend time retooling this, though.